core/src/auctions.ts

```diff
@@ -127,7 +135,42 @@ export const enrichAuctionWithPriceDrop = async function (auction: Auction): Pro
     }
 };
 
-export const enrichAuctionWithPriceDropAndMarketValue = async function (
+export const enrichAuctionWithMarketData = async function (auction: Auction, network: string): Promise<Auction> {
+    if (!auction.isActive || auction.isFinished || !auction.collateralToCoverDebt || !auction.approximateUnitPrice) {
+        return auction;
+    }
+    try {
+        const marketData = await getMarketData(network, auction.collateralSymbol, auction.collateralToCoverDebt);
+        const suggestedMarketId = (await getBestMarketData(marketData)).marketId;
+        for (const market of Object.values(marketData)) {
+            market.marketUnitPriceToUnitPriceRatio = auction.approximateUnitPrice
+                .minus(market.marketUnitPrice)
+                .dividedBy(market.marketUnitPrice);
+
+            market.transactionGrossProfit = calculateTransactionGrossProfit(
+                market.marketUnitPrice,
+                auction.collateralToCoverDebt,
+                auction.approximateUnitPrice
+            );
+
+            market.transactionGrossProfitDate = calculateTransactionGrossProfitDate(
+                auction,
+                market.marketUnitPrice,
+                await getNetworkDate(auction.network)
+            );
+        }
```

Comment:
You're again, modifying `market` object, without copying it which will have side effects